File: docs/versioned_docs/version-0.17/ci/quickstart/snippets/test/java/src/main/java/io/dagger/modules/hellodagger/HelloDagger.java
package io.dagger.modules.hellodagger;

import io.dagger.client.Container;
import io.dagger.client.DaggerQueryException;
import io.dagger.client.Directory;
import io.dagger.client.CacheVolume;
import io.dagger.module.annotation.DefaultPath;
import io.dagger.module.annotation.Function;
import io.dagger.module.annotation.Object;
import java.util.List;
import java.util.concurrent.ExecutionException;

/** HelloDagger main object */
@Object
public class HelloDagger {
  /** Return the result of running unit tests */
  @Function
  public String test(@DefaultPath("/") Directory source)
      throws InterruptedException, ExecutionException, DaggerQueryException {
        // get the build environment container
        // by calling another Dagger Function
    return this
        .buildEnv(source)
        // call the test runner
        .withExec(List.of("npm", "run", "test:unit", "run"))
        // capture and return the command output
        .stdout();
  }
}

File: docs/versioned_docs/version-0.17/ci/quickstart/snippets/test/php/src/HelloDagger.php
<?php

declare(strict_types=1);

namespace DaggerModule;

use Dagger\Attribute\DaggerFunction;
use Dagger\Attribute\DaggerObject;
use Dagger\Attribute\DefaultPath;
use Dagger\Attribute\Doc;
use Dagger\Container;
use Dagger\Directory;

use function Dagger\dag;

#[DaggerObject]
class HelloDagger
{
    #[DaggerFunction]
    #[Doc('Return the result of running unit tests')]
    public function test(
      #[DefaultPath('/')]
      Directory $source,
    ): string {
        return $this
            // get the build environment container
            // by calling another Dagger Function
            ->buildEnv($source)
            // call the test runner
            ->withExec(['npm', 'run', 'test:unit', 'run'])
            // capture and return the command output
            ->stdout();
    }
}
